framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -620,23 +729,43 @@ private void handleInitializers(
 
             // A list of arrayLens.  arrayLenOfDimensions.get(i) is the array lengths for the ith
             // dimension.
-            List<List<Integer>> arrayLenOfDimensions = new ArrayList<>();
+            List<RangeOrListOfValues> arrayLenOfDimensions = new ArrayList<>();
             for (ExpressionTree init : initializers) {
                 AnnotatedTypeMirror componentType = getAnnotatedType(init);
                 int dimension = 0;
                 while (componentType.getKind() == TypeKind.ARRAY) {
-                    List<Integer> arrayLens;
+                    RangeOrListOfValues rolv = null;
                     if (dimension == arrayLenOfDimensions.size()) {
-                        arrayLens = new ArrayList<>();
-                        arrayLenOfDimensions.add(arrayLens);
-                    } else {
-                        arrayLens = arrayLenOfDimensions.get(dimension);
+                        arrayLenOfDimensions.add(new RangeOrListOfValues());
                     }
                     AnnotationMirror arrayLen = componentType.getAnnotation(ArrayLen.class);
                     if (arrayLen != null) {
+                        rolv = arrayLenOfDimensions.get(dimension);
                         List<Integer> currentLengths = getArrayLength(arrayLen);
-                        arrayLens.addAll(currentLengths);
+                        if (rolv.isRange) {
```

Comment:
The method addAll in RangeOrList should take a list and properly add if RangeOrList is a range.